siac/parse.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+var errUnableToParseSize = errors.New("unable to parse size")
+
+// parseSize converts strings of form 10GB to a size in bytes. Fractional sizes
+// are truncated at the byte size. If parsing fails an error is returned.
+func parseSize(strSize string) (string, error) {
+	units := []struct {
+		suffix     string
+		multiplier int64
+	}{
+		{"kb", 1000},
+		{"mb", 1000000},
+		{"gb", 1000000000},
+		{"tb", 1000000000000},
+		{"kib", 1024},
+		{"mib", 1048576},
+		{"gib", 1073741824},
+		{"tib", 1099511627776},
```

Comment:
Ah, yes. I forgot about those.

I tried 10^3, 10^6, 2^10, 2^20, but that's bitwise xor in go so didn't work